.github/workflows/nightly.yml

```
@@ -38,17 +48,24 @@ jobs:
       - name: Check out code using Git
         uses: actions/checkout@v2
       
-      - name: Set Node version to 16
+      - name: Setup PNPM
+        uses: pnpm/action-setup@v2.0.1
+        with:
+          version: 6.23.6
+
+      - name: Setup Node
         uses: actions/setup-node@v2
         with:
           node-version: 16
-          cache: 'yarn'
+          cache: 'pnpm'
       
       - name: Clear lockfile
-        run: rm -rf yarn.lock node_modules 
+        run: rm -rf pnpm-lock.yaml node_modules 
       
-      - name: Install dependencies
-        run: yarn install --ignore-engines --ignore-scripts
+      - name: Install NPM Dependencies
+        run: pnpm install
+        env:
+          CI: true
```

Maybe, you could use [`pnpm up -r`](https://pnpm.io/cli/update) instead of removing the lockfile. This will also update the versions inside all package.json

```suggestion
      - name: Upgrade recursive
        run: pnpm up --recursive
```

---

Thanks! I wasn't able to commit your suggestion directly but I did add it in [d760b50](https://github.com/withastro/astro/pull/2455/commits/d760b5050c80a53a9ca0c309083c1c156135fd8b)